cmd/query/app/server.go

```diff
@@ -59,17 +60,17 @@ func NewServer(querySvc *querysvc.QueryService,
 	tm *tenancy.Manager,
 	telset telemetery.Setting,
 ) (*Server, error) {
-	_, httpPort, err := net.SplitHostPort(options.HTTPHostPort)
+	_, httpPort, err := net.SplitHostPort(options.HTTP.Endpoint)
 	if err != nil {
 		return nil, fmt.Errorf("invalid HTTP server host:port: %w", err)
 	}
-	_, grpcPort, err := net.SplitHostPort(options.GRPCHostPort)
+	_, grpcPort, err := net.SplitHostPort(options.GRPC.NetAddr.Endpoint)
 	if err != nil {
 		return nil, fmt.Errorf("invalid gRPC server host:port: %w", err)
 	}
 	separatePorts := grpcPort != httpPort || grpcPort == "0" || httpPort == "0"
 
-	if (options.TLSHTTP.Enabled || options.TLSGRPC.Enabled) && !separatePorts {
+	if (options.HTTP.TLSSetting != nil || options.GRPC.TLSSetting != nil) && !separatePorts {
```

Comment:
is it ever nil? During initialization you assign it unconditionally.

Reply:
I think we want `!TLSSetting.Insecure`

Reply:
Addressed in https://github.com/jaegertracing/jaeger/pull/6023#discussion_r1779497173